src/placement/jump_map_versions.c

```diff
@@ -317,15 +319,19 @@ __get_target_v1(struct pool_domain *root_pos, struct pool_domain *curr_pd,
 						      start_dom, end_dom, exclude_new);
 			if (range_set) {
 				if (top == -1) {
-					if (curr_pd != root_pos) {
-						/* all domains within the PD are full, ignore the
-						 * PD restrict.
+					if (curr_pd != root_pos && grp_size > 1) {
```

Comment:
let's say if there are 2 failure domain under one PD, and PD_number = 2,  then creating the object RP2_G2, so this should be created under 1 PD or multiple PD?

Reply:
I just tested the case -
With 2 server groups (PDs),
     2 servers for each group,
     1 engines for each server,
     1 targets for each engine,
     4 targets totalty (each server group with 2 targets).

layout of OC_RP_2G2 object, PDA 2 ---
[  0=>  3_PD1   1=>  2_PD1][  2=>  1_PD0   3=>  0_PD0]

layout of OC_RP_2G2 object, PDA 1 -
[  0=>  3_PD1   1=>  1_PD0][  2=>  2_PD1   3=>  0_PD0]
the number means -
[shard_idx => tgt_id_PD_idx]

Reply:
I mean if there are 2 targets per engine, then RP2_G2 will be placed to 2PD, instead of 1PD? is this expected for PD_number = 2.

Reply:
yes, will be placed to 2 PDs, the test result is -

With 2 server groups (PDs),
     2 servers for each group,
     1 engines for each server,
     2 targets for each engine,
     8 targets totalty (each server group with 4 targets).

layout of OC_RP_2G2 object, PDA 2 ...
[  0=>  7_PD1   1=>  5_PD1][  2=>  2_PD0   3=>  0_PD0]

layout of OC_RP_2G2 object, PDA 1 ...
[  0=>  7_PD1   1=>  2_PD0][  2=>  5_PD1   3=>  0_PD0]

looks that is good? or what kind of other layout you expect?

Reply:
Oh, I thought PD_number = 2 means the it will put 2 replicas in one PD, so in this case, I thought all shards will be put to 1 PD? or I misunderstand the PD_number ?

Reply:
no, PD_number is the number of PD( group), PDA (PD affinity) is the number of shards in one PD, I think the "PD number" in your mind is PDA.
Even for PDA = 2 case, for RP2G2, the 2 shards in the grp 0 will be put in one PD, but the 2 shards in grp 1 should be put in another PD.

Reply:
>Even for PDA = 2 case, for RP2G2, the 2 shards in the grp 0 will be put in one PD, but the 2 shards in grp 1 should be put in another PD.

Ah. Sure, thanks.

> If you want to put RP2G2's 4 shards in one PD, you should set PDA = 4
but a special thing is we only have 2 fault doms in one PD, the actual PDA was changed to 2 (see jmop->jmop_pd_grp_size init code in jm_obj_pd_init()).

Hmm, I thought in this case if PDA=4, then all shards should be in 1 PD? since different group can be put to the same domain anyway?

Reply:
> > If you want to put RP2G2's 4 shards in one PD, you should set PDA = 4
> > but a special thing is we only have 2 fault doms in one PD, the actual PDA was changed to 2 (see jmop->jmop_pd_grp_size init code in jm_obj_pd_init()).
> 
> Hmm, I thought in this case if PDA=4, then all shards should be in 1 PD? since different group can be put to the same domain anyway?

for RP4G1, should put to 2 PDs in that case. so for RP2G2, now did not differentiate and also put to 2 PDs